simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/admission/Admittor.java

```diff
@@ -38,10 +40,41 @@
    */
   boolean admit(long candidateKey, long victimKey);
 
+  /** Records the access to the entry. (overloading to use events)*/
+  default void record(AccessEvent event) { record(event.key()); }
+
+  /**
+   * Returns if the candidate should be added to the cache and the page replacement policy's chosen
+   * victim should be removed.
+   *
+   * @param candidate the event of the newly added entry
+   * @param victim the event of the entry the policy recommends removing
+   * @return if the candidate should be added and the victim removed due to eviction
+   */
+  default boolean admit(AccessEvent candidate, AccessEvent victim) { return admit(candidate.key(),victim.key()); }
+
   /** Returns an admittor that admits every candidate. */
   static Admittor always() {
     return AlwaysAdmit.INSTANCE;
   }
+
+  /** An admission policy for the cache that uses the AccessEvent object as input instead of key,
+   * useful for admission policies that are based on size or penalties.
+   * **/
+  interface EventsAdmittor extends Admittor {
```

Comment:
Maybe we should flip this around? The admittor was written prior to having an AccessEvent, where we introduced `KeyOnly` types for conversion. So we could make `KeyOnlyAdmittor` as a convenience for existing usages, while making the base interface event-aware.

Reply:
If you're okay with this proposal, I can go back and migrate towards that. That way it doesn't dirty up your PR and be an isolated refactoring.

Reply:
Sounds good, I'm good with your proposal.
Thanks